src/data/hooks/useRawFriendsDataFromFirebase.ts

```diff
@@ -0,0 +1,213 @@
+    LoadingStateError | LoadingStateCustom | null
+  >(null);
+  useEffect(() => {
+    if (!isAuthEnabled) return undefined;
+
+    const removeFriendsSnapshotListener = friendsCollection
+      .doc(account.id)
+      .onSnapshot(
+        {
+          // Ignore metadata changes
+          includeMetadataChanges: false,
+        },
+        (doc) => {
+          const data = doc.data();
+          if (data == null) {
+            setFriendData({ type: 'nonExistant' });
+          } else {
+            setFriendData({
+              type: 'exists',
+              data: doc.data() as FriendsData,
+            });
+          }
+        }
+      );
+    return (): void => {
+      removeFriendsSnapshotListener();
+    };
+  }, [account.id]);
+
+  const setFriendDataPersistent = useCallback(
+    (
+      next: ((current: FriendsData | null) => FriendsData | null) | FriendsData
+    ): void => {
+      let nextFriendData;
+      if (typeof next === 'function') {
+        let currentFriendData;
+        if (friendData.type === 'exists') {
+          currentFriendData = friendData.data;
+        } else {
+          currentFriendData = null;
+        }
+        nextFriendData = next(currentFriendData);
+      } else {
+        nextFriendData = next;
+      }
+      if (nextFriendData === null) return;
+
+      // Eagerly set the friend data here as well.
+      // It would be okay to wait until Firebase updates the state for us,
+      // (which it will do, even before the network calls are made),
+      // but this allows a window where state can react based on stale state.
+      setFriendData({ type: 'exists', data: nextFriendData });
+
+      friendsCollection
+        .doc(account.id)
+        .set(nextFriendData)
+        .catch((err) => {
+          softError(
+            new ErrorWithFields({
+              message: 'error when updating remote document',
+              source: err,
+              fields: {
+                accountId: account.id,
+              },
+            })
+          );
+        });
+    },
+    [account.id, friendData]
+  );
+
+  // Perform a transaction if the type is non-existent,
+  // trying to pull existing data from local storage
+  // and storing it in Firebase.
+  // This serves to provide the initial account data.
+  useEffect(() => {
+    if (!isAuthEnabled) return;
+
+    if (friendData.type === 'nonExistant') {
+      // Imperatively get the latest migrated data
+      let currentFriendsData: Immutable<FriendsData>;
+      try {
+        currentFriendsData = getFriendData() ?? defaultFriendsData;
+      } catch (err) {
+        softError(
+          new ErrorWithFields({
+            message:
+              'failed to imperatively get current friend data during account seeding',
+            source: err,
+            fields: {
+              accountId: account.id,
+            },
+          })
+        );
+        currentFriendsData = defaultFriendsData;
+      }
```

Comment:
Since we will not be loading friends data from local storage, we only need to initialize a default "friendsData" instance for each user, which is just line 143. This block of code can be reduced to
```
if (friendData.type === 'nonExistant') {
   currentFriendsData = defaultFriendsData;
}
```